Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/Generator.cs

```diff
@@ -175,9 +195,17 @@ public void Execute(GeneratorExecutionContext context)
                     // report every generated file to build output
                     diagnosticReporter.Report(Diagnostic.Create(DiagnosticDescriptors.CodeGeneration, Location.None, $"{model.GeneratedMethod.ContainingType.Name}.g.cs", sourceText));
 
+                    lambdaModels.Add(model);
                     annotationReport.LambdaFunctions.Add(model);
                 }
 
+                if (receiver.IsExecutable)
+                {
+                    var executableAssembly = new ExecutableAssembly(lambdaModels, lambdaModels[0].LambdaMethod.ContainingNamespace);
```

Comment:
We also need to add a diagnostic error if the project is missing a reference to `Amazon.Lambda.RuntimeSupport`. Otherwise users will get confused when they add the new executable attribute and they start getting weird compiler errors. You can find some examples in this library where we check to make sure their is dependency to the API Gateway event package if they are creating an API Gateway based function.